devs/docs/release.rst

```diff
@@ -55,10 +55,15 @@ This can happen either in master during the feature freeze, or after the switch
 to a release branch.
 
 
-Tagging a release
-=================
+Creating a new release
+======================
+
+Before creating a new distribution, some maintenance steps should get executed
+and a new version and tag should be created.
 
-Before creating a new distribution, a new version and tag should be created:
+- Invoke ``./gradlew generateLicenseNotes`` in order to update the
+  ``3RD-PARTY-NOTICES.md`` file, including licenses of all third-party
+  dependencies.
```

Comment:
I'd prefer if we can avoid making this another step for releases but instead run this on CI and fail if there would be a change.

So it gets updated if we change the dependencies. (And would also notice early if we introduced something with GPL (without dual-licene or classpath exception))

Reply:
I restored the original version with 57ad0daf52. It would be sweet to have a short conversation in order to give me some guidance about your plan to bring this to CI instead and how we would approach that.